### PR TITLE
Rename config.example to config.dev and rm config.test

### PR DESCRIPTION
**Pull request by marpio**
Renamed the config.example.toml -> config.dev.toml
I've also deleted config.test.toml since config.dev.toml can be used instead

### CODE REVIEW

File: cmd/olympus/actions/actions_test.go [hunk 1]
```diff
@@ -15,7 +15,7 @@ import (
 )
 
 var (
-	testConfigFile = filepath.Join("..", "..", "..", "config.test.toml")
```
Comment (michalpristas) [inline cmd/olympus/actions/actions_test.go]: i'm thinking about keeping it test in test and .dev. as a default in a code

Comment (marwan-at-work) [reply cmd/olympus/actions/actions_test.go]: @michalpristas I don't mind us deleting `config.test.toml` unless there's something about the test.toml file that is different from the regular one? We can always introduce it later on if the case turns out to be true.

Comment (marpio) [reply cmd/olympus/actions/actions_test.go]: At the moment they are identical.

Comment (michalpristas) [reply cmd/olympus/actions/actions_test.go]: if they're identical then one of them is redundant

Comment (marwan-at-work) [reply cmd/olympus/actions/actions_test.go]: yep, so let's merge this and if we ever need a special config values for tests we can put this file back, but i'd argue that we can just pass those special vals directly in the tests.

File: docs/content/walkthrough.md [hunk 2]
```diff
@@ -59,7 +59,7 @@ $ mkdir -p $(go env GOPATH)/src/github.com/gomods
 $ cd $(go env GOPATH)/src/github.com/gomods
 $ git clone https://github.com/gomods/athens.git
 $ cd athens
-$ GO111MODULE=off go run ./cmd/proxy -config_file=config.example.toml &
+$ GO111MODULE=off go run ./cmd/proxy -config_file=config.dev.toml &
```
Comment (marwan-at-work) [inline docs/content/walkthrough.md]: ops just realized we didn't need this flag anymore. will pr it after.

**Pull request merged**